### ISSUE TITLE
PrivateBin submits the link twice to YOURLS

### ISSUE THREAD

**Issue by P8ntBal1551** [PrivateBin/PrivateBin#825]
## Steps to reproduce

1. Create PrivateBin link
2. Click Shorten URL

### What happens

This JSON error
{"status":"fail","code":"error:url","url":{"keyword":"hwaxa","url":"https:\/\/privatebin.MYDOMAIN.com\/?655b4c43a3e8135c#Dm4RhgixjWGRyxDZLy6wGgzTTreQRY6PPzunU7pQd9DH","title":"https:\/\/privatebin.MYDOMAIN.com\/?655b4c43a3e8135c#Dm4RhgixjWGRyxDZLy6wGgzTTreQRY6PPzunU7pQd9DH","date":"2021-08-08 18:18:02","ip":"192.168.1.201","clicks":"0"},"message":"https:\/\/privatebin.MYDOMAIN.com\/?655b4c43a3e8135c#D[...] **already exists in database**","title":"https:\/\/privatebin.MYDOMAIN.com\/?655b4c43a3e8135c#Dm4RhgixjWGRyxDZLy6wGgzTTreQRY6PPzunU7pQd9DH","shorturl":"https:\/\/yourls.MYDOMAIN.com\/hwaxa","statusCode":200}

### What should happen

A page with a shortened URL

## Additional information

Here is the log from my yourls implementation

```
traefikIPAddress - - [08/Aug/2021:18:18:01 +0000] "GET /yourls-api.php?signature=mysignaure&action=shorturl&format=json&url=https%3A%2F%2Fprivatebin.mydomain.com%2F%3F655b4c43a3e8135c%23Dm4RhgixjWGRyxDZLy6wGgzTTreQRY6PPzunU7pQd9DH HTTP/1.1" 200 776 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:90.0) Gecko/20100101 Firefox/90.0"
traefikIPAddress - - [08/Aug/2021:18:18:02 +0000] "GET /yourls-api.php?signature=mysignaure&action=shorturl&format=json&url=https%3A%2F%2Fprivatebin.mydomain.com%2F%3F655b4c43a3e8135c%23Dm4RhgixjWGRyxDZLy6wGgzTTreQRY6PPzunU7pQd9DH HTTP/1.1" 200 814 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:90.0) Gecko/20100101 Firefox/90.0"
```

Keep in mind, I only pressed the button once

Here is my config with the comments removed

```
[main]name = "PrivateBin"
discussion = false
opendiscussion = false
password = true
fileupload = false
burnafterreadingselected = falsedefaultformatter = "plaintext"sizelimit = 10485760
template = "bootstrap-dark"languageselection = false
urlshortener = "https://yourls.MYDOMAIN.com/yourls-api.php?signature=MYSIGNATURE&action=shorturl&format=json&url="qrcode = true[expire]
default = "1day"[expire_options]
5min = 300
10min = 600
1hour = 3600
1day = 86400
1week = 6048001month = 2592000
1year = 31536000
never = 0[formatter_options]plaintext = "Plain Text"
syntaxhighlighting = "Source Code"
markdown = "Markdown"[traffic]
limit = 10
dir = PATH "data"[purge]limit = 300
batchsize = 10
dir = PATH "data"[model]
class = Filesystem
[model_options]
dir = PATH "data"
```

## Basic information

**Server address**: Private Server

**Server OS:** Docker Image

**Webserver:** Whatever is on the Docker Image

**Browser:** Firefox 90

**PrivateBin version:** 1.3.5

I can reproduce this issue on <https://privatebin.net>: No (No shortener)

**Comment by P8ntBal1551** [comment]
any progress on this?

**Comment by elrido** [comment]
I don't think anyone is working on this. Since you have a way of reproducing it as you are using a shortener, would you take this on, please? I personally don't use the shortener feature on any of my systems, so I have no way of investigating this.

**Comment by P8ntBal1551** [comment]
As the last time I just JS was....13? years ago, and I have never used AJAX, I could be totally off base with this. I don't really want to fork it just for 1 line of text, but from what I understand the fix is in js/privatebin.js; line 2042 should be taken out and line 2043 should change from

`.fail(function(data, textStatus, errorThrown) {`

to

`error: function(data, textStatus, errorThrown) {`

which I adapted from this stack overflow

https://stackoverflow.com/questions/41025392/ajax-success-failure-callback-not-working

**Comment by elrido** [comment]
I'm happy to raise the MR for you, if you can help us figure out what YOURLS is doing here. I don't think changing the function would be correct, as the doc at https://api.jquery.com/jquery.ajax/ states that in the promise returned by `.ajax()` we should use the `.fail()` instead of `.error()` methods:

> Deprecation Notice: The `jqXHR.success()`, `jqXHR.error()`, and `jqXHR.complete()` callbacks are removed as of jQuery 3.0. You can use `jqXHR.done()`, `jqXHR.fail()`, and `jqXHR.always()` instead.

Your logs show both responses being HTTP code 200, so successes from the point of view of the server. Your stack overflow link does suggest a possible reason for the first request failing to parse being that the response is not JSON. On line 2044 we would log the error reason to the browsers console log. Could you check what your browser logs there and share it here? You can open the browsers console using the [F12] key on Linux and Windows (Firefox, Chrome and derivatives).

**Comment by elrido** [comment]
Likely resolved via #997
